st3/lsp_utils/server_resource_interface.py

```diff
@@ -0,0 +1,69 @@
+from .helpers import SemanticVersion
+from abc import ABCMeta
+from abc import abstractmethod
+from abc import abstractproperty
+from LSP.plugin.core.typing import Dict
+
+__all__ = ['ServerStatus', 'ServerResourceInterface']
+
+
+class ServerStatus():
+    """
+    A :class:`ServerStatus` enum for use as a return value from :func:`ServerResourceInterface.get_status()`.
+    """
+
+    UNINITIALIZED = 1
+    """Initial status of the server."""
+    ERROR = 2
+    """Initiallation or update has failed."""
+    READY = 3
+    """Server is ready to provide resources."""
+
+
+class ServerResourceInterface(metaclass=ABCMeta):
```

Comment:
Should add some generic implementation that reports installation progress in ST3.

Reply:
That was handled in the client handler actually so I don't have a use case for adding a generic server resource implementation at the moment.